lib/core/services/synchronization_service.dart

```diff
@@ -286,6 +288,83 @@ class SynchronizationService {
     return await TeiRelatioShipOfflineProvider().getAllTeiRelationShips();
   }
 
+  Future<int> getOfflineEventsCount(CurrentUser currentUser) async {
+    int eventsCount = 0;
+    for (String orgUnit in currentUser.userOrgUnitIds) {
+      for (String program in currentUser.programs) {
+        int count =
+            await EventOfflineProvider().getOfflineEventCount(program, orgUnit);
+        eventsCount += count;
+      }
+    }
+    return eventsCount;
+  }
+
+  Future<int> getOfflineEnrollmentCount(CurrentUser currentUser) async {
+    int enrollmentsCount = 0;
+    for (String orgUnit in currentUser.userOrgUnitIds) {
+      for (String program in currentUser.programs) {
+        int count = await EnrollmentOfflineProvider()
+            .getOfflineEnrollmentsCount(program, orgUnit);
+        enrollmentsCount += count;
+      }
+    }
+    return enrollmentsCount;
+  }
+
+  Future<int> getTotalFromPaginator(String url,
+      {Map<String, dynamic> queryParameters}) async {
+    Response response = await httpClient.httpGetPagination(
+      url,
+      queryParameters,
+    );
+    Map<String, dynamic> pager = json.decode(response.body)['pager'];
```

Comment:
try to wrap under try and catch in case of any fails

Reply:
Noted.

Reply:
This is handled